src/http_server.c

```diff
@@ -0,0 +1,313 @@
+/*
+ *  Copyright (c) 2025 Rainer Keller
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "config.h"
+#include "http.h" // for url_encode
```

Comment:
Either we document why we include headers for all of them, or for none of them. We have chosen the latter so far:
```suggestion
#include "http.h"
 ```